java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/explorer/DefaultDecorationsImpl.java

```diff
@@ -110,23 +116,52 @@ public TreeItemData createDecorations(Node n, boolean expanded) {
             }
 
             FileObject f = n.getLookup().lookup(FileObject.class);
+            boolean nodeChecked = false;
+            if (f == null) {
+                DataFolder df = n.getLookup().lookup(DataFolder.class);
+                if (df != null) {
+                    f = df.getPrimaryFile();
+                    // Workaround for possible bug in data folder: DataFolder itself and the Fileobject for the folder do not
+                    // have a Node in their Lookups.
```

Comment:
Is it Folder node or Package node? I believe @entlicher (when working on version control system integration?) tried hard to not treat Package node as regular `FileObject`, but rather a "flat" folder.

Reply:
The bug was found on the `package` node, which indeed does not have any FileObject (directly) in its Lookup. But it has a DataFolder in it ... and the `DataFolder` has its `primaryFile` property. During debugging it turned out (IIRC) that `DataFolder.getLookup().lookup(Node.class) == null` as well as `primaryFile.getLookup().lookup(Node.class) == null`, which is different from file-based DataObjects.

Reply:
If it is a bug, can it be fixed?

Reply:
It should be - but I need to prioritize given my limited capacity: tracked as [NETBEANS-6296
(https://issues.apache.org/jira/browse/NETBEANS-6294)